pkg/kando/location.go

```diff
@@ -38,17 +50,86 @@ func newLocationCommand() *cobra.Command {
 	cmd.AddCommand(newLocationDeleteCommand())
 	cmd.PersistentFlags().StringP(pathFlagName, "s", "", "Specify a path suffix (optional)")
 	cmd.PersistentFlags().StringP(profileFlagName, "p", "", "Pass a Profile as a JSON string (required)")
-	_ = cmd.MarkFlagRequired(profileFlagName)
+	cmd.PersistentFlags().StringP(repositoryServerFlagName, "r", "", "Pass a Repository Server CR as a JSON string (required for kopia based blueprints)")
 	return cmd
 }
 
+// NewDataMover creates an instance of DataMover Interface and returns
+// the preferred DataMover as per the arguments passed in kando command
+func NewDataMover(dm *DataMoverArgs, outputName, snapJson string) datamover.DataMover {
+	switch dm.Type {
+	case DataMoverTypeProfile:
+		return datamover.NewProfileDataMover(dm.Profile, outputName, snapJson)
+	case DataMoverTypeRepositoryServer:
+		return datamover.NewRepositoryServerDataMover(dm.RepositoryServer, outputName, snapJson)
+	default:
+		return nil
+	}
+}
+
 func pathFlag(cmd *cobra.Command) string {
 	return cmd.Flag(pathFlagName).Value.String()
 }
 
+// validateCommandArgs makes sure that we are getting exactly
+// one of --profile or --repository-server flags
+func validateCommandArgs(cmd *cobra.Command) error {
+	profileFlag := cmd.Flags().Lookup(profileFlagName).Value.String()
+	repositoryServerFlag := cmd.Flags().Lookup(repositoryServerFlagName).Value.String()
+	if profileFlag != "" && repositoryServerFlag != "" {
+		return errors.New("Either --profile or --repository-server can be provided")
+	}
+	if profileFlag == "" && repositoryServerFlag == "" {
+		return errors.New("Please provide either --profile or --repository-server as per the datamover you want to use")
+	}
+	return nil
+}
+
+func dataMoverFromCMD(cmd *cobra.Command, flag string) (datamover.DataMover, error) {
+	dm := &DataMoverArgs{}
+	profile := cmd.Flags().Lookup(profileFlagName).Value.String()
+	if profile != "" {
+		profileRef, err := unmarshalProfileFlag(cmd)
+		if err != nil {
+			return nil, err
+		}
+		dm = &DataMoverArgs{
+			Type:    DataMoverTypeProfile,
+			Profile: profileRef,
+		}
+	}
+	repositoryServer := cmd.Flags().Lookup(repositoryServerFlagName).Value.String()
+	if repositoryServer != "" {
+		repositoryServerRef, err := unmarshalRepositoryServerFlag(cmd)
+		if err != nil {
+			return nil, err
+		}
+		dm = &DataMoverArgs{
+			Type:             DataMoverTypeRepositoryServer,
+			RepositoryServer: repositoryServerRef,
+		}
+	}
```

Comment:
Lets move these snippet to two different helper function - something like `dataMoverArgsForProfile()` and `dataMoverArgsForRepoServer()`

Reply:
We can also create a helper method which returns `DataMoverType` from `cmd.Flags`. And add switch case to call correct `dataMoverArgsFor*` function.

The function should do only one thing. Break this function into multiple ones.

Reply:
Okay will implement it !

Reply:
Done !